packages/hooks/use-fills.test.tsx

```diff
@@ -0,0 +1,321 @@
+    delete store.state.fills;
+
+    const Comp = connect(() => {
+      const fills = useFills("slot 1");
+      expect(fills).toEqual([]);
+
+      return (
+        <>
+          {fills.map(({ Fill, props, key }) => (
+            <Fill key={key} {...props} />
+          ))}
+        </>
+      );
+    });
+
+    const app = create(
+      <Provider value={store}>
+        <Comp />
+      </Provider>
+    );
+
+    expect(app.toJSON()).toEqual(null);
+    expect(app.toJSON()).toMatchSnapshot();
+  });
+
+  it("Should re-render the fill when updating the props", () => {
+    const Comp = connect(() => {
+      const fills = useFills("slot 1");
+
+      return (
+        <>
+          {fills.map(({ Fill, props, key }) => (
+            <Fill key={key} {...props} />
+          ))}
+        </>
+      );
+    });
+
+    const app = create(
+      <Provider value={store}>
+        <Comp />
+      </Provider>
+    );
+
+    act(() => {
+      store.actions.fillActions.setNumber(43);
+    });
+
+    expect(app.toJSON().props).toEqual({ id: "test-fill", "data-number": 43 });
+    expect(app.toJSON().children[0]).toEqual("Im a Fill");
+
+    expect(app.toJSON()).toMatchSnapshot();
+  });
+
+  it("Should render the fills in the order of priority", () => {
+    store.state.fills["more important fill"] = {
+      slot: "slot 1",
+      library: "FillComponent",
+      priority: 1,
+      props: {
+        number: 43,
+      },
+    };
+
+    const Comp = connect(() => {
+      const fills = useFills("slot 1");
+
+      return (
+        <>
+          {fills.map(({ Fill, props, key }) => (
+            <Fill key={key} {...props} />
+          ))}
+        </>
+      );
+    });
+
+    const app = create(
+      <Provider value={store}>
+        <Comp />
+      </Provider>
+    );
+
+    // This fill should come first
+    expect(app.toJSON()[0].props).toEqual({
+      id: "test-fill",
+      "data-number": 43,
+    });
+
+    // This fill should come second
+    expect(app.toJSON()[1].props).toEqual({
+      id: "test-fill",
+      "data-number": 42,
+    });
+
+    expect(app.toJSON()).toMatchSnapshot();
+  });
+
+  it("Should skip rendering the fills with value `false`", () => {
+    store.state.fills["other fill"] = {
+      slot: "slot 1",
+      library: "FillComponent",
+      props: {
+        number: 1,
+      },
+    };
+
+    store.state.fills["test fill"] = false;
+
+    const Comp = connect(() => {
+      const fills = useFills("slot 1");
+
+      return (
+        <>
+          {fills.map(({ Fill, props, key }) => (
+            <Fill key={key} {...props} />
+          ))}
+        </>
+      );
+    });
+
+    const app = create(
+      <Provider value={store}>
+        <Comp />
+      </Provider>
+    );
+
+    // We should only render 1 component with the prop `{number: 1}`
+    expect(app.toJSON().props).toEqual({
+      id: "test-fill",
+      "data-number": 1,
+    });
+
+    expect(app.toJSON()).toMatchSnapshot();
+  });
+});
```

Comment:
Great tests.

The only one I miss is to check if it doesn't include a fill that doesn't match the slot name.